### PR TITLE
Use Apple's aggregate audio device where available

### PR DESCRIPTION
**Pull request by danra**
I also considered removing `AudioIODeviceCombiner` completely but didn't, mainly because adding the appropriate error handling or disabling aggregate devices on older platforms instead of just crashing seems to require some more work and refactoring of the existing JUCE code.

### CODE REVIEW

File: modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp [hunk 1]
```diff
@@ -947,10 +947,32 @@ class CoreAudioIODevice   : public AudioIODevice,
...
+                        @(kAudioSubDeviceDriftCompensationQualityKey) : @(kAudioSubDeviceDriftCompensationMaxQuality),
+                    },
+                    @{
+                        @(kAudioSubDeviceUIDKey) : getDeviceUID (outputDeviceId),
+                        @(kAudioSubDeviceDriftCompensationKey) : @(1),
+                        @(kAudioSubDeviceDriftCompensationQualityKey) : @(kAudioSubDeviceDriftCompensationMaxQuality),
+                    },
+                ],
+            };
+            OSStatus status = AudioHardwareCreateAggregateDevice ((CFDictionaryRef)description, &aggregateDeviceID);
```
Comment (yairchu) [inline modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp:973]: Perhaps this merits a comment explaining that we wouldn't get here unless `createDevice` verified that this API is available.

File: modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp [hunk 2]
```diff
@@ -2152,7 +2187,13 @@ class CoreAudioIODeviceType  : public AudioIODeviceType,
         auto combinedName = outputDeviceName.isEmpty() ? inputDeviceName
                                                        : outputDeviceName;
 
-        if (inputDeviceID == outputDeviceID)
+        // Newer Apple platforms can create aggregate audio devices
+        bool canCreateAppleAggregateDevice = [] {
```
Comment (yairchu) [inline modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp:2192]: Why is this in a lambda?

**Review by yairchu — APPROVED** [review]
Wrote some minor suggestions/questions, if this is tested to also work normally on macOS < 10.9 then LGTM :)

File: modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp (hunk 2 above)
Comment (danra) [reply modules/juce_audio_devices/native/juce_mac_CoreAudio.cpp:2192]: If I use `@available` and not `if (@available...` Xcode warns that it's not actually guarding availability. Given that, this seemed the right way to me

**Pull request merged**